codec-http/src/test/java/io/netty/handler/codec/http/HttpHeadersTest.java

```diff
@@ -76,8 +76,8 @@ public void testAddSelf() {
         headers.add(headers);
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testSetSelf() {
+    @Test
```

Comment:
This is definitely a change in semantic and so worthy of discussion. 

I think the problem in the original code is the inconsistency in behavior between this.setAll(this) and this.set(this), one would throw and the other would not.
- these APIs are relatively easy for users to confuse 
- if the preconditions were dropped they would have the same outcome, they are both idempotent for this case
- the behavior was not documented

I suspect exposure to the semantic difference is rare, it would be odd to see this occur and odder still to be depending on it throwing for some real conditional behavior